From c8dd98fc3a82acd69dac8e88aa67332b4d52834a Mon Sep 17 00:00:00 2001
From: Nikita Yushchenko <nikita.yoush@cogentembedded.com>
Date: Thu, 25 May 2023 10:42:30 +0600
Subject: [PATCH 008/157] net: ethernet: renesas: rswitch: turn rtsn_ptp into a
 separate module

This is required for proper sharing of rtsn_ptp between rtsn and
rswitch, and support configs other than everything built into the
kernel.

Signed-off-by: Nikita Yushchenko <nikita.yoush@cogentembedded.com>

diff --git a/drivers/net/ethernet/renesas/Kconfig b/drivers/net/ethernet/renesas/Kconfig
index 497a5874a6c3..04acc0333c44 100644
--- a/drivers/net/ethernet/renesas/Kconfig
+++ b/drivers/net/ethernet/renesas/Kconfig
@@ -42,12 +42,17 @@ config RAVB
 	  This driver supports the following SoCs:
 		- R8A779x.
 
+config RTSN_PTP
+	tristate
+	default n
+
 config RENESAS_ETHER_SWITCH
 	tristate "Renesas Ethernet Switch support"
 	depends on ARCH_RENESAS || COMPILE_TEST
 	select CRC32
 	select MII
 	select PHYLIB
+	select RTSN_PTP
 	help
 	  Renesas Ethernet Switch device driver.
 	  This driver supports the following SoCs:
@@ -59,6 +64,7 @@ config RTSN
         select CRC32
         select MII
         select PHYLIB
+	select RTSN_PTP
         help
           Renesas Ethernet-TSN device driver.
           This driver supports the following SoCs:
diff --git a/drivers/net/ethernet/renesas/Makefile b/drivers/net/ethernet/renesas/Makefile
index d52ae998fc6b..1848c370a555 100644
--- a/drivers/net/ethernet/renesas/Makefile
+++ b/drivers/net/ethernet/renesas/Makefile
@@ -9,8 +9,6 @@ ravb-objs := ravb_main.o ravb_ptp.o
 
 obj-$(CONFIG_RAVB) += ravb.o
 
+obj-$(CONFIG_RTSN_PTP) += rtsn_ptp.o
+obj-$(CONFIG_RTSN) += rtsn.o
 obj-$(CONFIG_RENESAS_ETHER_SWITCH) += rswitch.o
-
-rtsn_drv-objs := rtsn.o rtsn_ptp.o
-
-obj-$(CONFIG_RTSN) += rtsn_drv.o
diff --git a/drivers/net/ethernet/renesas/rtsn_ptp.c b/drivers/net/ethernet/renesas/rtsn_ptp.c
index ceb000b13726..314bd44dd9aa 100644
--- a/drivers/net/ethernet/renesas/rtsn_ptp.c
+++ b/drivers/net/ethernet/renesas/rtsn_ptp.c
@@ -163,6 +163,7 @@ int rtsn_ptp_init(struct rtsn_ptp_private *ptp_priv, enum rtsn_ptp_reg_layout la
 
 	return 0;
 }
+EXPORT_SYMBOL_GPL(rtsn_ptp_init);
 
 int rtsn_ptp_unregister(struct rtsn_ptp_private *ptp_priv)
 {
@@ -175,6 +176,7 @@ int rtsn_ptp_unregister(struct rtsn_ptp_private *ptp_priv)
 
 	return 0;
 }
+EXPORT_SYMBOL_GPL(rtsn_ptp_unregister);
 
 struct rtsn_ptp_private *rtsn_ptp_alloc(struct platform_device *pdev)
 {
@@ -188,3 +190,8 @@ struct rtsn_ptp_private *rtsn_ptp_alloc(struct platform_device *pdev)
 
 	return ptp;
 }
+EXPORT_SYMBOL_GPL(rtsn_ptp_alloc);
+
+MODULE_AUTHOR("Phong Hoang <phong.hoang.wz@renesas.com>");
+MODULE_DESCRIPTION("Renesas Gen4 PTP subdriver");
+MODULE_LICENSE("GPL v2");
-- 
2.25.1

